cmd/zfs/zfs_main.c

```diff
@@ -1512,6 +1550,13 @@ zfs_do_destroy(int argc, char **argv)
 	at = strchr(argv[0], '@');
 	pound = strchr(argv[0], '#');
 	if (at != NULL) {
+		if (types_specified && (types & ZFS_TYPE_SNAPSHOT) == 0) {
+			(void) fprintf(stderr, gettext("tried to destroy a "
+			    "snapshot, but -t did not allow snapshots to be "
+			    "destroyed.\n"));
```

Comment:
It would be nice to include the snapshot/bookmark/dataset name in the error message and to try and keep it fairly concise to match the existing style.  How about, `"cannot destroy '%s'; restricted by the -t option.\n"`, `argv[0]` will contain the name,